src/vk/surface.cc

```diff
@@ -0,0 +1,51 @@
+#include <cassert>
+
+#include "common/glfw_include.h"
+#include "common/vulkan_include.h"
+
+#include "vkwrap/surface.h"
+
+using namespace vkwrap;
+
+vk::SurfaceKHR
+Surface::createSurface( vk::Instance instance, GLFWwindow* window )
+{
+    assert( instance != nullptr );
```

Comment:
If you expect window parameter to be non null, then you should pass it by reference.

Reply:
Truth.